pkg/policy/l4.go

```diff
@@ -230,12 +230,16 @@ func (l4 L4Filter) matchesLabels(labels labels.LabelArray) bool {
 
 // L4PolicyMap is a list of L4 filters indexable by protocol/port
 // key format: "port/proto"
-type L4PolicyMap map[string]L4Filter
+type L4PolicyMap struct {
+	Filters          map[string]L4Filter
+	SourceRuleLabels labels.Labels // The User rule labels that resulted in this Map. Can be empty
+	SourceRuleCount  int64         // Number of rules that resulted in this Map
+}
 
 // HasRedirect returns true if at least one L4 filter contains a port
 // redirection
 func (l4 L4PolicyMap) HasRedirect() bool {
```

Comment:
Since `L4PolicyMap` is no longer a map, should we change the receiver to a pointer? Answer: Not on this case. But are there any other cases where there are methods that use attributes of `L4PolicyMap` and need a receiver as a pointer?

Reply:
You are correct overall, though. I'll change them all out to be consistent (it also means no one can accidentally call them on a non-pointer)